pkg/machinery/config/types/v1alpha1/generate/generate.go

```diff
@@ -254,45 +256,45 @@ func NewSecretsBundleFromConfig(c config.Provider) *SecretsBundle {
 }
 
 // NewEtcdCA generates a CA for the Etcd PKI.
-func NewEtcdCA() (ca *x509.CertificateAuthority, err error) {
+func NewEtcdCA(currentTime time.Time) (ca *x509.CertificateAuthority, err error) {
 	opts := []x509.Option{
 		x509.RSA(true),
 		x509.Organization("etcd"),
-		x509.NotAfter(time.Now().Add(87600 * time.Hour)),
+		x509.NotAfter(currentTime.Add(87600 * time.Hour)),
 	}
 
 	return x509.NewSelfSignedCertificateAuthority(opts...)
 }
 
 // NewKubernetesCA generates a CA for the Kubernetes PKI.
-func NewKubernetesCA() (ca *x509.CertificateAuthority, err error) {
+func NewKubernetesCA(currentTime time.Time) (ca *x509.CertificateAuthority, err error) {
 	opts := []x509.Option{
 		x509.RSA(true),
 		x509.Organization("kubernetes"),
-		x509.NotAfter(time.Now().Add(87600 * time.Hour)),
+		x509.NotAfter(currentTime.Add(87600 * time.Hour)),
 	}
 
 	return x509.NewSelfSignedCertificateAuthority(opts...)
 }
 
 // NewTalosCA generates a CA for the Talos PKI.
-func NewTalosCA() (ca *x509.CertificateAuthority, err error) {
+func NewTalosCA(currentTime time.Time) (ca *x509.CertificateAuthority, err error) {
 	opts := []x509.Option{
 		x509.RSA(false),
 		x509.Organization("talos"),
-		x509.NotAfter(time.Now().Add(87600 * time.Hour)),
+		x509.NotAfter(currentTime.Add(87600 * time.Hour)),
```

Comment:
we should probably here and in other places explicitly set `x509.NotBefore(currentTime)`

Reply:
(or whatever the default is in x509 package)

Reply:
fixed